interfaces/builtin/mount_control_test.go

```diff
@@ -0,0 +1,283 @@
+	restore := systemd.MockSystemdVersion(208, nil)
+	defer restore()
+	err := interfaces.BeforeConnectPlug(s.iface, s.plug)
+	c.Assert(err, ErrorMatches, `systemd version 208 is too old \(expected at least 209\)`)
+}
+
+func (s *MountControlInterfaceSuite) TestSanitizePlugUnhappy(c *C) {
+	var mountControlYaml = `name: consumer
+version: 0
+plugs:
+ mntctl:
+  interface: mount-control
+  %s
+apps:
+ app:
+  plugs: [mntctl]
+`
+	data := []struct {
+		plugYaml      string
+		expectedError string
+	}{
+		{
+			"", // missing "mount" attribute
+			`mount-control "mount" attribute must be a list of dictionaries`,
+		},
+		{
+			"mount: a string",
+			`mount-control "mount" attribute must be a list of dictionaries`,
+		},
+		{
+			"mount: [this, is, a, list]",
+			`mount-control "mount" attribute must be a list of dictionaries`,
+		},
+		{
+			"mount:\n  - what: [this, is, a, list]\n    where: /media/**",
+			`mount-control "what" must be a string`,
+		},
+		{
+			"mount:\n  - what: /path/\n    where: [this, is, a, list]",
+			`mount-control "where" must be a string`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /\n    persistent: string",
+			`mount-control "persistent" must be a boolean`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /\n    type: string",
+			`mount-control "type" must be an array of strings.*`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /\n    type: [true, false]",
+			`mount-control "type" element 1 not a string.*`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/*\n    type: [auto)]",
+			`mount-control filesystem type invalid.*`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/*\n    type: [upperCase]",
+			`mount-control filesystem type invalid.*`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/*\n    type: [two words]",
+			`mount-control filesystem type invalid.*`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/*\n",
+			`mount-control "options" cannot be empty`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /\n    options: string",
+			`mount-control "options" must be an array of strings.*`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/*\n    options: []",
+			`mount-control "options" cannot be empty`,
+		},
+		{
+			"mount:\n  - what: here\n    where: /",
+			`mount-control "what" attribute is invalid: must start with / and not contain special characters`,
+		},
+		{
+			"mount:\n  - what: /double\"quote\n    where: /",
+			`mount-control "what" attribute is invalid: must start with / and not contain special characters`,
+		},
+		{
+			"mount:\n  - what: /variables/are/not/@{allowed}\n    where: /",
+			`mount-control "what" attribute is invalid: must start with / and not contain special characters`,
+		},
+		{
+			"mount:\n  - what: /invalid}pattern\n    where: /",
+			`mount-control "what" setting cannot be used: Invalid closing brace, no matching open.*`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /path",
```

Comment:
Is it? I'm asking because it's not a trivial thing to check: the plug could declare something like

    what: /{something/,}

which would pass our regexp check, but still allow just `/` to be specified. Well, we could change our regexp to something like `^/[^{}*][^"@]*$`, but I'm not sure it's needed.

Reply:
yeah we would break things if we allow snaps to mount things on `/`, I see your point though that since we are allowing a regex a simple check to disallow `/` is not sufficient. I guess it's fine as-is then, it would be more complicated to attempt to validate that any regexp doesn't match `/` somehow. I suppose we could compile the AARE into a Go regexp and then see if it matches some directories we know we don't want to allow, but that would be more work with questionable benefit considering that we do always still have this gated on the snap upload review process anyways.